dxp/exporting.md

```diff
@@ -0,0 +1,47 @@
+# Exporting your JavaScript
+
+This document describes how you should organize you npm packages to export JavaScript functionality
+
+## Introduction
+
+Although you usually consume packages via [npm](https://npmjs.com), you are very likely to create packages yourself: for example if you want to expose useful functionality and be able to reuse it elsewhere.
+
+## Declaring your module
+
+All the information related to your (`npm`) package, lives in a `package.json` file: as it name indicates, this file is written in JSON and needs to be valid.
+
+If you don't see a `package.json` file in the root of your OSGi module, you can create it by hand or by running the following command
```

Comment:
I also thought about all the "stuff" that `npm init` adds to `package.json` and didn't really know if it was a good idea to "encourage" users to use it ... so I'll rework that.

I could point to an existing `package.json` in DXP and say that obivously the name and version must be changed.

I remember correctly in DXP the `version` in `package.json` needs to have the same value as the `version` of the `bnd.bnd file`. 

Here's an example of that for our beloved `frontend-js-web` "thing" (I'm not calling that a package anymore :joy: ) , here's the [`package.json`](https://github.com/liferay/liferay-portal/blob/b4c82067fd9450bf1574d98335afa00f65172cf5/modules/apps/frontend-js/frontend-js-web/package.json#L44) and the [`bnd.bnd`](https://github.com/liferay/liferay-portal/blob/b4c82067fd9450bf1574d98335afa00f65172cf5/modules/apps/frontend-js/frontend-js-web/bnd.bnd#L3)

Reply:
> I'm not calling that a package anymore

I call them "modules" because they correspond to OSGi modules; they just happen to have some JS and JSON stowing away inside them.

Reply:
Well that makes sense, and it will help me describe what these "things" are - thanks!